scripts/get_objf_and_derivs_split.py

```diff
@@ -351,6 +354,57 @@ def MergeAndComputeObjfForSplit(split_index):
     MergeAllOrders(split_index)
     ComputeObjfAndFinalDerivs(split_index, args.derivs_out != None)
 
+def CleanUpIfNeeded():
+    # For cleaning up intermediate files that won't be used, note if option need-model = true
+    # be used, clean up will keep file 'float.all'
+    if args.clean_up == 'true':
+        try:
+            if not os.path.isdir(args.work_dir):
+                ExitProgram("error finding working directory {0}".format(args.work_dir))
+            # If need_model option set to be false, this script will keep "float.all" in
+            # each split dir for future use
+            need_split_float = True if args.need_model == 'false' else False
+            for split_index in range(1, args.num_splits + 1):
```

Comment:
I found where the ‘a’s come from
In get_counts.py, line 332:

test_command = "bash -c 'set -o pipefail; (echo a; echo b) | "\
        "distribute-input-lines /dev/stdout /dev/null’";

maybe, we can change it to,

test_command = "bash -c 'set -o pipefail; (echo a; echo b) | "\
        "distribute-input-lines 2 > /dev/null’";

A peculiar point I cannot understand is that why it only print ‘a’, not ‘b’, e.g.
when I change it to 

test_command = "bash -c 'set -o pipefail; (echo a; echo b; echo c; echo d; echo e; echo f;) | "\
        "distribute-input-lines /dev/stdout /dev/null'";

get_counts.py: dumping counts
a
c
e

only the 1st, 3st, and  5st thing in “echo” got printed, this is somewhat strange




On Jul 12, 2016, at 3:17 PM, Daniel Povey <notifications@github.com> wrote:

>  I'm new to python- try google.